## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

There are more than 3.04 million apps found on Google Play Store. With this project we will analyzing various apps found on the play store with the help of different python libraries.

# It consists of 13 columns:- App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, and Android Ver with 10841 Rows.

**Imports:** Let us start by importing some of the required libraries with which we will be working on.

In [3]:
# Imports
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')



# Loading the dataset as a pandas data frame.

In [4]:
googlestore_df =  pd.read_csv('/content/Play Store Data.csv')

In [5]:
googlestore_df.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

In [6]:
googlestore_df.tail(10)

App  ...         Android Ver
10831                       payermonstationnement.fr  ...          4.0 and up
10832                                       FR Tides  ...          2.1 and up
10833                                    Chemin (fr)  ...          2.2 and up
10834                                  FR Calculator  ...          4.1 and up
10835                                       FR Forms  ...          4.0 and up
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10 rows x 13 columns]

After loading the dataset, we can start the exploration but before that, we need to check and see that the dataset is ready for performing several exploration operations or not, so let’s first have a look at the structure and the manner in which the data is organized.

# To know if there is any missing value or Nan value in the dataset, we can use the isnull() function.

In [7]:
googlestore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

So, we will need to prepare the dataset before performing exploratory data analysis on it.










We saw that the dataset contains many Null or missing values. The column Rating, Type , Content Rating , Current Ver , andAndroid Ver contains 1474, 1, 1, 8, and 3 missing values respectively.


Will it not be better if we can define a function to get more useful information about the different attributes of the dataset, also there is one more valid point in defining a function which it will be reusable, and we are going to utilize our defined function several times in future


In [8]:
def printinfo():
    temp = pd.DataFrame(index=googlestore_df.columns)
    temp['data_type'] = googlestore_df.dtypes
    temp['null_count'] = googlestore_df.isnull().sum()
    temp['unique_count'] = googlestore_df.nunique()
    return temp

Let’s call the function and see what it returns:

In [9]:
printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

We have some useful information about the dataset. i.e., we can now see the missing number of values of any attribute, its unique count, and its respective data types.
Now we can start the process of data cleaning, lets start with the column Type :-

In [10]:
googlestore_df[googlestore_df.Type.isnull()]

App  ...         Android Ver
9148  Command & Conquer: Rivals  ...  Varies with device

[1 rows x 13 columns]

In [11]:
googlestore_df['Type'].fillna("Free", inplace = True)

After filling the value we can check and see if that has been correctly placed.

In [12]:
googlestore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64